check-oracle.yml

```diff
@@ -0,0 +1,309 @@
+          register: oratab_exists
+
+        - name: oratab not found
+          fail: 
+            msg: "The file {{ oratab_path }} was not found on the target"
+          when: not ( oratab_exists.stat.exists | bool )
+
+        - name: Slurp oratab from remote
+          slurp:
+            src: "{{ oratab_path }}"
+          register: oratab_contents
+
+        - name: Split oratab file contents into lines
+          set_fact:
+            oratab_lines: "{{ oratab_contents.content | b64decode | split('\n') }}"
+
+        - name: Check for ORACLE_SID
+          shell: "grep ^{{ ORACLE_SID }} {{ oratab_path }}"
+          register: oratab_results
+          ignore_errors: true
+
+        - name: ORACLE_HOME Not Found
+          fail: 
+            msg: "ORACLE_HOME was not found on the target for {{ ORACLE_SID }}"
+          when: oratab_results.rc != 0
+
+        - name: Parse oratab to find Oracle Home for {{ ORACLE_SID }}
+          set_fact:
+            ORACLE_HOME: >-
+              {{
+                (oratab_lines
+                 | select('match', '^' ~ ORACLE_SID ~ ':')
+                 | list
+                 | first
+                ).split(':')[1]
+              }}
+          when: oratab_lines is defined and (oratab_lines | length) > 0
+          failed_when: oratab_lines is not defined
+
+        - name: Display ORACLE_HOME
+          debug:
+            msg: "Found ORACLE_HOME={{ ORACLE_HOME }}"
+
+        - name: Get a tmpfile name
+          set_fact:
+            TMPFILE: "{{ lookup('pipe','mktemp') }}"
+
+        - name: Run ORAchk
+          become: true
+          shell: "ORACLE_PATH={{ orachk_script_dir }} SQLPATH={{ orachk_script_dir }} {{ ORACHK_BASE }}/orachk/orachk -s -dbconfig {{ ORACLE_HOME }}%{{ ORACLE_SID }} -showpass -profile dba | tee {{ TMPFILE }}"
+          environment:
+            ORACLE_SID: "{{ ORACLE_SID }}"
+            PATH: "{{ extended_path }}"
+          args:
+            chdir: "{{ orachk_script_dir }}"
+
+        - name: Get ORAchk tmpfile status
+          stat:
+            path: "{{ TMPFILE }}"
+          register: tmpfile_location
+
+        - name: Check that ORAchk tmpfile exists
+          fail:
+            msg: "Could not find ORAchk tmpfile {{ TMPFILE }}"
+          when: not tmpfile_location.stat.exists
+
+        - name: Retrieve ORAchk zipfile name from tmpfile
+          shell: "grep UPLOAD {{ TMPFILE }} | awk '{ print $NF }'"
+          register: orachk_file_info
+
+        - name: Get a tmpfile name
+          set_fact:
+            ORACHK_RPT_FILE: "{{ orachk_file_info.stdout }}"
+
+        - name: Verify ORAchk Report File Existence
+          stat:
+            path: "{{ ORACHK_RPT_FILE }}"
+          register: orachk_rpt_location
+
+        - name: Was ORAchk report name found
+          fail:
+            msg: "Could not locate ORAchk report file {{ ORACHK_RPT_FILE }}"
+          when: not orachk_rpt_location.stat.exists
+
+        - name: Fetch the ORAchk zipfile
```

Comment:
I'm getting permission errors here too:

TASK [Fetch the ORAchk zipfile] ***************************************************************************************
fatal: [mfielding-ora122-node1]: FAILED! => {"changed": false, "msg": "file is not readable: /opt/oracle.ahf/data/mfielding-ora122-node1/orachk/orachk_mfielding-ora122-node1_040325_17063.zip"}

$ sudo ls -l /opt/oracle.ahf/data/mfielding-ora122-node1/orachk/orachk_mfielding-ora122-node1_040325_17063.zip
-r--r----- 1 root root 3902854 Apr  3 17:11 /opt/oracle.ahf/data/mfielding-ora122-node1/orachk/orachk_mfielding-ora122-node1_040325_17063.zip

Reply:
I had thought it might be necessary to use root to get this file, but it was not needed in my environment for some reason.
I will change that, and set ownership appropriately for the orachk zip file.